pkg/logbridge/logbridge.go

```diff
@@ -6,50 +6,80 @@ import (
 	"io"
 	"time"
 
+	"github.com/square/p2/Godeps/_workspace/src/github.com/rcrowley/go-metrics"
 	"github.com/square/p2/pkg/logging"
 )
 
-// Copy implements a buffered copy operation between dest and src.
+type LogBridge struct {
+	Reader        io.Reader
+	DurableWriter io.Writer
+	LossyWriter   io.Writer
+
+	logger        logging.Logger
+	metrics       MetricsRegistry
+	logLinesCount metrics.Counter
+	logBytes      metrics.Counter
+}
+
+type MetricsRegistry interface {
+	Register(metricName string, metric interface{}) error
+}
+
+func NewLogBridge(r io.Reader,
+	durableWriter io.Writer,
+	lossyWriter io.Writer,
+	logger logging.Logger,
+	metricsRegistry MetricsRegistry,
+	loglineMetricKeyName string,
```

Comment:
There is probably a reasonable middle ground in letting this project define metric suffixes and letting the Square side provide a single prefix - that would get us the best of both worlds presumably